### PR TITLE
[Issue- #22] Abstraction to work with Entity:  Native query API

### PR DESCRIPTION
**Pull request by olehhaliak**
This is the first PR related to issue #22 Abstraction to work with an entity.
Query API allows users to create custom native queries, and will be used as a base to build repository abstraction later on

### CODE REVIEW

File: src/main/java/com/bobocode/blyznytsia/bibernate/exception/MissingParamsException.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.bobocode.blyznytsia.bibernate.exception;
+
+import com.bobocode.blyznytsia.bibernate.query.Query;
+
+/**
+ * Exception thrown during {@link   Query} execution in case when one or more parameters
```
Comment (vitshch) [inline src/main/java/com/bobocode/blyznytsia/bibernate/exception/MissingParamsException.java]: Some extra spaces between @link and Query

File: src/main/java/com/bobocode/blyznytsia/bibernate/query/Query.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  Optional<T> getSingleResult();
+
+  /**
+   * Executes the query and returns a list of results.
+   *
+   * @return a {@code List} of results, or an empty {@code List} if there are no results
+   * @throws MissingParamsException if any of the query parameters were not set
+   */
+  List<T> getResultList();
+}
```
Comment (vitshch) [inline src/main/java/com/bobocode/blyznytsia/bibernate/query/Query.java]: Add newline at the end of the file

File: src/test/java/com/bobocode/blyznytsia/bibernate/query/TypedQueryTest.java [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+    @Test
+    void throwsMissingParamsExceptionIfParameterIsNotSet() {
+      var sql = "SELECT * FROM sample WHERE id= :paramThatWillNotBeSet";
+      var typedQuery = new TypedQuery<>(entityType, sql,mockEntityPersister);
+      assertThrows(MissingParamsException.class, typedQuery::getResultList);
+    }
+  }
+
+}
```
Comment (vitshch) [inline src/test/java/com/bobocode/blyznytsia/bibernate/query/TypedQueryTest.java]: Add newline at the end

**Review by vitshch — CHANGES_REQUESTED** [review]

File: src/main/java/com/bobocode/blyznytsia/bibernate/query/Query.java (hunk 2 above)
Comment (olehhaliak) [reply src/main/java/com/bobocode/blyznytsia/bibernate/query/Query.java]: fixed

File: src/test/java/com/bobocode/blyznytsia/bibernate/query/TypedQueryTest.java (hunk 3 above)
Comment (olehhaliak) [reply src/test/java/com/bobocode/blyznytsia/bibernate/query/TypedQueryTest.java]: fixed

File: src/main/java/com/bobocode/blyznytsia/bibernate/exception/MissingParamsException.java (hunk 1 above)
Comment (olehhaliak) [reply src/main/java/com/bobocode/blyznytsia/bibernate/exception/MissingParamsException.java]: fixed

File: src/test/java/com/bobocode/blyznytsia/bibernate/query/TypedQueryTest.java (hunk 3 above)
Comment (olehhaliak) [reply src/test/java/com/bobocode/blyznytsia/bibernate/query/TypedQueryTest.java]: fixed

**Review by marynamelnichuk — APPROVED** [review]
I hope unit tests will be in the next PR.

**Review by vitshch — APPROVED** [review]

**Pull request merged**